cli/internal/mcp/utils/processDocuments.go

```diff
@@ -0,0 +1,195 @@
+	systemURI := &DocumentURI{
+		SystemID: system.ID,
+	}
+	fmt.Fprintf(&results.Result, "  <system id=\"%s\">\n", systemURI.String())
+	results.Result.WriteString("    <documentation>\n")
+
+	processDocumentation(results, documentationData, system, filter, includeContent)
+
+	results.Result.WriteString("    </documentation>\n")
+	results.Result.WriteString("  </system>\n")
+}
+
+func processDocumentation(results *Results, documentationData *DocumentationData, system *sqlite.System, filter *DocumentURI, includeContent bool) {
+	if documentationSources, exists := documentationData.Documentation[system.ID]; exists {
+		// Sort documentation source IDs alphabetically
+		documentationIDS := make([]string, 0, len(documentationSources))
+		for documentationID := range documentationSources {
+			documentationIDS = append(documentationIDS, documentationID)
+		}
+		sort.Strings(documentationIDS)
+
+		for _, documentationID := range documentationIDS {
+			documentation := documentationSources[documentationID]
+			if filter == nil || filter.DocumentationID == "" || filter.DocumentationID == documentation.ID {
+				processDocumentationSource(results, documentationData, documentation, filter, includeContent)
+			}
+		}
+	}
+}
+
+func processDocumentationSource(results *Results, documentationData *DocumentationData, documentation *sqlite.SystemDocumentation, filter *DocumentURI, includeContent bool) {
+	documentationURI := &DocumentURI{
+		SystemID:        documentation.SystemID,
+		DocumentationID: documentation.ID,
+	}
+	fmt.Fprintf(&results.Result, "      <documentation_source id=\"%s\">\n", documentationURI.String())
+	results.Result.WriteString("        <documents>\n")
+
+	processDocuments(results, documentationData, documentation, filter, includeContent)
+
+	results.Result.WriteString("        </documents>\n")
+	results.Result.WriteString("      </documentation_source>\n")
+}
+
+func processDocuments(results *Results, documentationData *DocumentationData, documentation *sqlite.SystemDocumentation, filter *DocumentURI, includeContent bool) {
+	if documents, exists := documentationData.Documents[documentation.SystemID][documentation.ID]; exists {
+		// Sort document paths alphabetically
+		documentPaths := make([]string, 0, len(documents))
+		for documentPath := range documents {
+			documentPaths = append(documentPaths, documentPath)
+		}
+		sort.Strings(documentPaths)
+
+		for _, documentPath := range documentPaths {
+			document := documents[documentPath]
+			if filter == nil || filter.DocumentPath == "" || strings.HasPrefix(document.ID, filter.DocumentPath) {
+				processDocument(results, documentationData, document, includeContent)
+				results.Total++
+			}
+		}
+	}
+}
+
+// generateSourceURL generates a source URL from a SystemDocumentation Path and document ID
+func generateSourceURL(basePath string, documentID string) string {
+	// Only try to parse as a git URL if basePath ends with .git
```

Comment:
Let's add a note in the review of this change to consider adding the extractor "type" to the DB so we can put this in a switch statement by type rather than inferring it.